From 3fd216d3f40bd91ee99d74674e045fd2a1cd6cc0 Mon Sep 17 00:00:00 2001
From: Zhou Jingyu <b02241@freescale.com>
Date: Wed, 19 Jan 2011 20:43:54 +0800
Subject: [PATCH] ENGR00138213-2 Add da9053 power key to mx53 smd &loco

add defconfig, remove gpio power key
fix some defconfig item miss

Signed-off-by: Zhou Jingyu <Jingyu.Zhou@freescale.com>
---
 arch/arm/configs/imx5_defconfig |   11 +++++++----
 arch/arm/mach-mx5/mx53_smd.c    |    1 -
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/arch/arm/configs/imx5_defconfig b/arch/arm/configs/imx5_defconfig
index 80391cb..53aa870 100644
--- a/arch/arm/configs/imx5_defconfig
+++ b/arch/arm/configs/imx5_defconfig
@@ -1,7 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Linux kernel version: 2.6.35.3
-# Thu Jan  6 18:20:36 2011
+# Wed Jan 19 20:30:04 2011
 #
 CONFIG_ARM=y
 CONFIG_HAVE_PWM=y
@@ -265,6 +265,7 @@ CONFIG_MACH_MX50_RDP=y
 #
 CONFIG_ARCH_MXC_HAS_NFC_V3=y
 CONFIG_ARCH_MXC_HAS_NFC_V3_2=y
+CONFIG_MXC_BLUETOOTH_RFKILL=y
 CONFIG_IRAM_ALLOC=y
 CONFIG_DMA_ZONE_SIZE=96
 CONFIG_ISP1504_MXC=y
@@ -527,7 +528,7 @@ CONFIG_BT_HIDP=y
 #
 CONFIG_BT_HCIBTUSB=y
 # CONFIG_BT_HCIBTSDIO is not set
-CONFIG_BT_HCIUART is not set
+CONFIG_BT_HCIUART=y
 # CONFIG_BT_HCIUART_H4 is not set
 # CONFIG_BT_HCIUART_BCSP is not set
 # CONFIG_BT_HCIUART_LL is not set
@@ -660,7 +661,6 @@ CONFIG_MTD_NAND_IDS=y
 # CONFIG_MTD_NAND_NANDSIM is not set
 # CONFIG_MTD_NAND_IMX_NFC is not set
 CONFIG_MTD_NAND_MXC_V3=y
-CONFIG_MXC_BLUETOOTH_RFKILL=y
 # CONFIG_MTD_NAND_MXC_SWECC is not set
 # CONFIG_MTD_NAND_MXC_FORCE_CE is not set
 # CONFIG_MXC_NAND_LOW_LEVEL_ERASE is not set
@@ -707,10 +707,11 @@ CONFIG_MISC_DEVICES=y
 # CONFIG_SENSORS_TSL2550 is not set
 # CONFIG_DS1682 is not set
 # CONFIG_TI_DAC7512 is not set
+CONFIG_MXS_PERFMON=y
 # CONFIG_ANDROID_PMEM is not set
 # CONFIG_UID_STAT is not set
 # CONFIG_C2PORT is not set
-CONFIG_MXS_PERFMON=y
+
 #
 # EEPROM support
 #
@@ -962,6 +963,7 @@ CONFIG_TOUCHSCREEN_MXC=y
 # CONFIG_TOUCHSCREEN_W90X900 is not set
 # CONFIG_TOUCHSCREEN_TPS6507X is not set
 CONFIG_TOUCHSCREEN_DA9052=y
+CONFIG_TOUCHSCREEN_MAX11801=y
 CONFIG_INPUT_MISC=y
 # CONFIG_INPUT_AD714X is not set
 # CONFIG_INPUT_ATI_REMOTE is not set
@@ -973,6 +975,7 @@ CONFIG_INPUT_MISC=y
 CONFIG_INPUT_UINPUT=y
 # CONFIG_INPUT_PCF8574 is not set
 # CONFIG_INPUT_GPIO_ROTARY_ENCODER is not set
+CONFIG_INPUT_DA9052_ONKEY=y
 
 #
 # Hardware I/O ports
diff --git a/arch/arm/mach-mx5/mx53_smd.c b/arch/arm/mach-mx5/mx53_smd.c
index 9472da7..9371432 100644
--- a/arch/arm/mach-mx5/mx53_smd.c
+++ b/arch/arm/mach-mx5/mx53_smd.c
@@ -765,7 +765,6 @@ static struct i2c_board_info mxc_i2c1_board_info[] __initdata = {
 }
 
 static struct gpio_keys_button smd_buttons[] = {
-	GPIO_BUTTON(MX53_SMD_PMIC_ON_OFF_REQ, KEY_POWER, 0, "power", 0),
 	GPIO_BUTTON(MX53_SMD_KEY_VOL_UP, KEY_VOLUMEUP, 1, "volume-up", 0),
 	GPIO_BUTTON(MX53_SMD_KEY_VOL_DOWN, KEY_VOLUMEDOWN, 1, "volume-down", 0),
 };
-- 
1.5.4.4

